SW-3666 Add planting site selector to observations page

karthikbtf opened the pull request:
- Plants dashboard and plants observations have the same page structure
- extracted out plants dashboard page into a wrapper page component
- reused that for observations
- added a concept of 'All' as a selection
- added a story for the page view

<img width="769" alt="PlantsPrimaryPageView - Default ⋅ Storybook 2023-05-26 10-09-10" src="https://github.com/terraware/terraware-web/assets/1865174/f2dd7584-936d-46e7-9642-9ac154e5b6b8">

<img width="764" alt="PlantsPrimaryPageView - Default ⋅ Storybook 2023-05-26 10-08-54" src="https://github.com/terraware/terraware-web/assets/1865174/565cc067-0e36-44a4-bb8e-522a30c2c3fd">

<img width="775" alt="Terraware App 2023-05-26 09-50-54" src="https://github.com/terraware/terraware-web/assets/1865174/c323f1d3-d373-48df-917f-af66e7910226">

charlie-tf commented on src/components/Observations/index.tsx:
```diff
@@ -1,3 +1,30 @@
+      title={strings.OBSERVATIONS}
+      onSelect={onSelect}
+      pagePath={APP_PATHS.PLANTING_SITE_OBSERVATIONS}
+      lastVisitedPreferenceName='lastObservationsPlantingSite'
```
Thoughts on namespacing these in a structured way to keep things organized? Such as `'<page>.<preferenceName>'`, e.g. `'observations.lastPlantingSite'`. Realize that this might break with how preferences have been organized in previous pages/components, if they are organized at all...

karthikbtf replied:
That's a good suggestion, I understand the structure. I prefer to keep preference names flatter for now, let each component decide how they want to name the preferences. Also yes, this will break existing preferences that are already in the database. The preference names are not parsed for semantics anywhere.

karthikbtf replied:
Here's a suggestion, since we aren't using plants dashboard in production yet.
I can make the preferences for the planting site pages hierarchical for sure. We can apply this going forward to other preferences.
For existing preferences (like accessions table, etc.) we will keep them as they are today.

charlie-tf replied:
Sounds lovely. Thank you!

charlie-tf commented on src/components/Plants/index.tsx:
```diff
@@ -1,153 +1,36 @@
-        />
-      </Box>
-    </TfMain>
+    <PlantsPrimaryPage
```
Nice consolidation

charlie-tf commented on src/components/PlantsPrimaryPage/index.tsx:
```diff
@@ -0,0 +1,184 @@
+  const setActivePlantingSite = useCallback(
+    (site: PlantingSite | undefined) => {
+      if (site) {
+        history.push(pagePath.replace(':plantingSiteId', site.id.toString()));
```
Should this be a `replace` type of history update? Do we want the user to go to the sequence of last selected planting sites when they hit the back button?

karthikbtf replied:
Yes we do want them to go back. The selections are url driven today.

charlie-tf reviewed (commented):
Nice work, just a couple suggestions

karthikbtf commented:
I am going to clean this up some more - see if i can create a story without any BE dependencies

karthikbtf commented:
> I am going to clean this up some more - see if i can create a story without any BE dependencies

Was able to refactor this, extracted out the rendering and created a story for that. Also updated the preference names.

charlie-tf reviewed (approved):
🎉
